### PR TITLE
feat(authentication): REST API-Based MQTT 5.0 Enhanced Authentication

### PR DESCRIPTION
**Pull request by Meggielqk**

### CODE REVIEW

File: dir.yaml [hunk 1]
```diff
@@ -269,6 +269,7 @@
             - access-control/authn/ldap
             - access-control/authn/http
         - access-control/authn/scram
+        - access-control/authn/scram_http
```
Comment (zmstone) [inline dir.yaml]:
```suggestion
        - access-control/authn/scram_restapi
```

**Review by zmstone — COMMENTED** [review]
scram-http.png files are better to be renamed too.

**Comment by id** [comment]
`client_attrs` should be moved back to the original PR. This feature is for 5.8, it does not exist in 5.7.

File: zh_CN/access-control/authn/scram_restapi.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
+# 基于 REST API 的 MQTT 5.0 增强认证
```
Comment (zmstone) [inline zh_CN/access-control/authn/scram_restapi.md]:
```suggestion
# 基于 REST API 的 MQTT 5.0 SCRAM 认证
```

File: en_US/access-control/authn/scram_restapi.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
+# REST API-Based MQTT 5.0 Enhanced Authentication
```
Comment (zmstone) [inline en_US/access-control/authn/scram_restapi.md]:
```suggestion
# REST API-Based MQTT 5.0 SCRAM Authentication
```

**Review by zmstone — APPROVED** [review]

**Pull request merged**